lib/oauth2/request.ex

```diff
@@ -7,6 +7,7 @@ defmodule OAuth2.Request do
   alias OAuth2.{Client, Error, Response}
 
   @type body :: any
+  @http_client Application.get_env(:oauth2, :http_client, OAuth2.HTTPClient.Hackney)
```

Comment:
Almost missed this one,

If we do it this way, people may need to know that they need to recompile things if a dependency changes. 

By now, I would suggest to remove it from module attributes into a function

```elixir
defp http_client
  Application.get_env(:oauth2, :http_client, OAuth2.HTTPClient.Hackney)
end
```